nl_server/store/memory.py

```diff
@@ -69,74 +54,58 @@ def __init__(self, embeddings_path: str, model: any) -> None:
     self.dataset_embeddings = torch.from_numpy(df.to_numpy()).to(torch.float)
 
   #
-  # Given a list of queries, searches the in-memory embeddings index
-  # and returns a map of candidates keyed by input queries.
+  # Given a list of query embeddings, searches the in-memory embeddings index
+  # and returns a list of candidates in the same order as original queries.
   #
-  def search_vars(self,
-                  queries: List[str],
-                  skip_topics: bool = False) -> Dict[str, Dict]:
-    query_embeddings = self.model.encode(queries, show_progress_bar=False)
-    top_k = _NUM_SV_INDEX_MATCHES
-    if skip_topics:
-      top_k = _NUM_SV_INDEX_MATCHES_WITHOUT_TOPICS
+  def vector_search(self, query_embeddings: torch.Tensor,
+                    top_k: int) -> List[vars.VarCandidates]:
     hits = semantic_search(query_embeddings,
                            self.dataset_embeddings,
                            top_k=top_k)
 
     # A map from input query -> SV DCID -> matched sentence -> score for that match
-    query2sv2sentence2score: Dict[str, Dict[str, Dict[str, float]]] = {}
+    query_indexed_sv2sentence2score: List[Dict[str, Dict[str, float]]] = []
     # A map from input query -> SV DCID -> highest matched score
-    query2sv2score: Dict[str, Dict[str, float]] = {}
+    query_indexed_sv2score: List[Dict[str, float]] = []
     for i, hit in enumerate(hits):
-      q = queries[i]
-      query2sv2score[q] = {}
-      query2sv2sentence2score[q] = {}
+      query_indexed_sv2sentence2score.append({})
+      query_indexed_sv2score.append({})
       for ent in hit:
         score = ent['score']
         for dcid in self.dcids[ent['corpus_id']].split(','):
-          if skip_topics and dcid.startswith(_TOPIC_PREFIX):
-            continue
           # Prefer the top score.
-          if dcid not in query2sv2score[q]:
-            query2sv2score[q][dcid] = score
-            query2sv2sentence2score[q][dcid] = {}
+          if dcid not in query_indexed_sv2score[i]:
+            query_indexed_sv2score[i][dcid] = score
+            query_indexed_sv2sentence2score[i][dcid] = {}
 
           if ent['corpus_id'] >= len(self.sentences):
             continue
           sentence = self.sentences[ent['corpus_id']]
-          query2sv2sentence2score[q][dcid][sentence] = score
+          query_indexed_sv2sentence2score[i][dcid][sentence] = score
 
-    query2result: Dict[str, vars.VarCandidates] = {}
+    results: List[vars.VarCandidates] = []
```

Comment:
curious why the change from a list to a dict?

Reply:
Because the input changed from query string to query embeddings, it seemed more natural to return a list corresponding to the input list.